tests/slurm_plugin/test_clustermgtd.py

```diff
@@ -1232,14 +1243,21 @@ def test_maintain_nodes(private_ip_to_instance_map, active_nodes, mock_unhealthy
     mock_handle_powering_down_nodes = mocker.patch.object(
         cluster_manager, "_handle_powering_down_nodes", auto_spec=True
     )
+    mock_handle_protected_mode_process = mocker.patch(
+        "slurm_plugin.clustermgtd.ClusterManager._handle_protected_mode_process"
+    )
     # Run test
-    cluster_manager._maintain_nodes(private_ip_to_instance_map, active_nodes)
+    cluster_manager._maintain_nodes(private_ip_to_instance_map, active_nodes, partitions)
     # Check function calls
     mock_update_replacement.assert_called_with(active_nodes)
     mock_find_unhealthy.assert_called_with(active_nodes, private_ip_to_instance_map)
     mock_handle_dynamic.assert_called_with(mock_unhealthy_nodes[0], private_ip_to_instance_map)
     mock_handle_static.assert_called_with(mock_unhealthy_nodes[1], private_ip_to_instance_map)
     mock_handle_powering_down_nodes.assert_called_with(active_nodes, private_ip_to_instance_map)
+    assert_that(mock_handle_protected_mode_process.call_count).is_equal_to(1)
```

Comment:
Can we add a test case to test that `handle_protected_mode_process` is not called when protected mode is disabled?